### PR TITLE
support for MPD_HOST

### PR DESCRIPTION
**Pull request by alicebob**
for #28

### CODE REVIEW

File: siren/main.go [hunk 1]
```diff
@@ -24,7 +25,8 @@ func main() {
 		os.Exit(0)
 	}
 
-	c, err := NewMPD(*mpdURL)
+	u := url(*mpdURL)
```
Comment (robx) [inline siren/main.go]: I'd expect the command line argument to override `MPD_HOST` if a command line argument is given. It doesn't seem to, though?

File: siren/main.go [hunk 2]
```diff
@@ -45,3 +48,13 @@ func mux(c *MPD, root http.FileSystem) *http.ServeMux {
 	r.Handle("/", http.FileServer(root))
 	return r
 }
+
+func url(u string) string {
+	if h, _ := os.LookupEnv("MPD_HOST"); h != "" {
```
Comment (robx) [inline siren/main.go]: let's do `MPD_PORT` also?

File: siren/main.go (hunk 1 above)
Comment (alicebob) [reply siren/main.go]: fair enough!

File: siren/main.go [hunk 3]
```diff
@@ -45,3 +48,26 @@ func mux(c *MPD, root http.FileSystem) *http.ServeMux {
...
+		host = h
+		if p, _ := os.LookupEnv("MPD_PORT"); p != "" {
+			port = p
+		}
+	}
+
+	if u != "" {
+		if h, p, err := net.SplitHostPort(u); err != nil {
+			host = u
+			port = "6600"
```
Comment (robx) [inline siren/main.go]: and leave port be here

File: siren/main.go [hunk 4]
```diff
@@ -45,3 +48,26 @@ func mux(c *MPD, root http.FileSystem) *http.ServeMux {
 	r.Handle("/", http.FileServer(root))
 	return r
 }
+
+func url(u string) string {
+	host, port := "localhost", "6600"
+
+	if h, _ := os.LookupEnv("MPD_HOST"); h != "" {
+		host = h
+		if p, _ := os.LookupEnv("MPD_PORT"); p != "" {
```
[the remark this answers is not in the document]
Comment (alicebob) [reply siren/main.go]: I had that before, but then you get:

`MPD_PORT=123 ./siren -mpd localhost`  -> connects to `localhost:123`

[the remark this answers is not in the document]
Comment (robx) [reply siren/main.go]: ah yes, that's also weeeeird

**Pull request merged**